snippets/fsharp/VS_Snippets_CLR_System/system.console.class/fs/example3.fs

```diff
@@ -0,0 +1,169 @@
+﻿
+// dump a range of Unicode characters as a 16x16 array
+// <Snippet4>
+open System
+open System.IO
+open System.Globalization
+open System.Text
+
+type uint = uint32
+
+let roundDownToMultipleOf (b:uint) (u:uint) : uint =
+    u - (u % b)
+
+let roundUpToMultipleOf (b:uint) (u:uint) : uint =
+    roundDownToMultipleOf b u |> (+) b
+
+let displayRange (start:uint) (``end``:uint) : unit =
```

Comment:
These three return type annotations aren't required.

Reply:
What do you think about the convention that if you specify explicit type annotations for any parameter or return value that you fully specify the types for the remaining part of the function signature? If I see something like the function definition below it looks to me at a glance like it has no return value.

```f#
let doSomethingWithTwoStrings (a:string) (b:string) =
    // implementation ...
```

Reply:
Generally we don't have that convention, since it's not necessary for these kinds of things. The only way the type would end up different is if you use a custom operator that did a cast, or casted yourself. It's also worth noting that these kinds of functions are also typically declared `inline` so that type inference doesn't specialize to a specific numeric type, letting you use it for `int`/`uint`/`float`/etc.

Reply:
I've removed the unnecessary annotations and made the first two functions inline.